daemon/src/engine/sys/con_tty.cpp

```diff
@@ -66,9 +67,27 @@ static struct termios TTY_tc;
 static Console::Field TTY_field(INT_MAX);
 
 void WriteToStdout(const char* text) {
-	if (write( STDOUT_FILENO, text, strlen(text)) < 0) {
-        Log::Warn("Error writing to the terminal: %s", strerror(errno));
-    }
+
+	int ret;
+
+	ret = write( STDOUT_FILENO, text, strlen(text) );
+
+	if ( ret < 0 && errno == EAGAIN )
+	{
+		fd_set set;
+		FD_ZERO( &set );
+		FD_SET( STDOUT_FILENO, &set );
+		select( STDOUT_FILENO + 1, NULL, &set, NULL, NULL);
```

Comment:
Either blocking or dropping is fine, with a slight preference for blocking because that only happens when there is a lot of console spam, which is rare.